internal/shaderir/program.go

```diff
@@ -74,16 +75,17 @@ type Block struct {
 }
 
 type Stmt struct {
-	Type        StmtType
-	Exprs       []Expr
-	Blocks      []*Block
-	ForVarType  Type
-	ForVarIndex int
-	ForInit     constant.Value
-	ForEnd      constant.Value
-	ForOp       Op
-	ForDelta    constant.Value
-	InitIndex   int
+	Type          StmtType
+	Exprs         []Expr
+	Blocks        []*Block
+	ForVarType    Type
+	ForVarIndex   int
+	ForInit       constant.Value
+	ForEnd        constant.Value
+	ForOp         Op
+	ForDelta      constant.Value
+	InitIndex     int
+	IsTypeGuessed bool
```

Comment:
If it is a short declaration `:=`, type is implicit.

Reply:
shaderir.Stmt itself doesn't have a type information, so having `IsTypeGuessed` here is odd.